src/awkward/operations/ak_concatenate.py

```diff
@@ -240,7 +242,7 @@ def action(inputs, depth, **kwargs):
                     all_flatten.append(f)
 
                 offsets = backend.index_nplike.empty(
-                    len(nextinputs[0]) + 1, dtype=np.int64
+                    nextinputs[0].length + 1, dtype=np.int64
```

Comment:
Good catch!

Reply:
> This looks good, although there's at least one use of `to_rectilinear` in the tests, which is where this is failing right now.
> 
> https://github.com/scikit-hep/awkward/blob/6cde4b3fd8f36b51fba202274e56e82cea129e9f/tests/test_1247_numpy_to_rectilinear_ndarray.py#L13

I could have sworn I ran the test suite locally. Oh, well, thank goodness for CI!

Reply:
It came up in the Linux-ROOT test; it might be a test that requires ROOT in the specific environment to run. Although I see that another one failed, too, so maybe not.